File: solutions/c/isogram/2/isogram.c
#include "isogram.h"

bool is_isogram(const char *p) {
    bool s[26] = {0};

    for (; p && *p; p++)
        if ((*p|32) >= 'a' && (*p|32) <= 'z' && (s[(*p|32) - 'a'] ? true : (s[(*p|32) - 'a'] = true, false)))
            return false;

    return p;
}

File: solutions/zig/isogram/3/isogram.zig
const std = @import("std");

pub fn isIsogram(str: []const u8) bool {
    var charsFound: [26]bool = [_]bool{false} ** 26;

    for (str) |char| {
        const lowChar: u8 = std.ascii.toLower(char);
        if (!std.ascii.isAlphabetic(lowChar)) continue;
        if (charsFound[lowChar - 'a']) return false;
        charsFound[lowChar - 'a'] = true;
    }

    return true;
}
